core/slim_sim.cpp

```diff
@@ -6416,9 +6419,23 @@ void SLiMSim::WritePopulationTable(tsk_table_collection_t *p_tables)
 		// first, write out empty entries for unused subpop ids before this one
 		while (last_id_written < subpop_id - 1)
 		{
-			tsk_id_t tsk_population = tsk_population_table_add_row(&p_tables->populations, (char *)&last_subpop_id, (uint32_t)0);	// the address is unused
-			if (tsk_population < 0) handle_error("tsk_population_table_add_row", tsk_population);
 			last_id_written++;
+			if (last_id_written < (slim_objectid_t) population_table_copy->num_rows) {
+				tsk_population_t tsk_population_object;
+				ret = tsk_population_table_get_row(population_table_copy, last_id_written, &tsk_population_object);
+				if (ret != 0) handle_error("WritePopulationTable tsk_population_table_get_row()", ret);
+				tsk_population_id = tsk_population_table_add_row(
+						&p_tables->populations,
+						tsk_population_object.metadata,
+						tsk_population_object.metadata_length);
+				if (tsk_population_id < 0) handle_error("tsk_population_table_add_row", tsk_population_id);
+			} else {
+				tsk_population_id = tsk_population_table_add_row(
+						&p_tables->populations,
+						NULL, 0);
```

Comment:
just to check: the old code passed `(char *)&last_subpop_id` instead of NULL when adding new empty rows.  is the switch to NULL definitely OK?

Reply:
NULL is what we usually use in the tskit code so unless C++ is for some reason different, then yes?

Reply:
See for instance here: https://github.com/tskit-dev/tskit/blob/f1fd80462ae1b0e211d651440b88c224b7cda393/c/tests/test_tables.c#L168